src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -20,11 +20,26 @@ import (
 // Actor methods
 ////////////////////////////////////////////////////////////////////////////////
 
-func (a *StorageMinerActorCode_I) GetWorkerVRFKey(rt Runtime) filcrypto.VRFPublicKey {
+func (a *StorageMinerActorCode_I) StageWorkerKeyChange(rt Runtime, key filcrypto.VRFPublicKey) (err error) {
 	h, st := a.State(rt)
-	ret := st.Info().WorkerVRFKey()
+	rt.ValidateImmediateCallerIs(st.Info().GetWorker(rt.CurrEpoch()))
```

Comment:
This must be the *owner*, not the worker, so the owner can recover from compromised or lost keys.

Reply:
Thanks for catching that!